src/token/common/erc2981/erc2981.cairo

```diff
@@ -0,0 +1,165 @@
+    use openzeppelin::introspection::src5::SRC5Component::SRC5Impl;
+    use openzeppelin::introspection::src5::SRC5Component;
+    use openzeppelin::token::common::erc2981::interface::{IERC2981, IERC2981_ID};
+
+    use starknet::ContractAddress;
+    use starknet::storage::Map;
+
+    #[derive(Serde, Drop, PartialEq, starknet::Store)]
+    struct RoyaltyInfo {
+        pub receiver: ContractAddress,
+        pub royalty_fraction: u256,
+    }
+
+    #[storage]
+    struct Storage {
+        default_royalty_info: RoyaltyInfo,
+        token_royalty_info: Map<u256, RoyaltyInfo>,
+    }
+
+    //
+    // External
+    //
+    #[embeddable_as(ERC2981Impl)]
+    impl ERC2981<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl SRC5: SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>,
+    > of IERC2981<ComponentState<TContractState>> {
+        /// Returns the receiver address and amount to send for royalty
+        /// for token id `token_id` and sale price `sale_price`
+        fn royalty_info(
+            self: @ComponentState<TContractState>, token_id: u256, sale_price: u256
+        ) -> (ContractAddress, u256) {
+            let royalty_info: RoyaltyInfo = self.token_royalty_info.read(token_id);
+            let mut royalty_receiver = royalty_info.receiver;
+            let mut royalty_fraction = royalty_info.royalty_fraction;
+
+            if royalty_receiver.is_zero() {
+                let default_royalty_info: RoyaltyInfo = self.default_royalty_info.read();
+                royalty_receiver = default_royalty_info.receiver;
+                royalty_fraction = default_royalty_info.royalty_fraction;
+            }
+
+            let royalty_amount: u256 = (sale_price * royalty_fraction) / self._fee_denominator();
+
+            (royalty_receiver, royalty_amount)
+        }
+    }
+
+
+    //
+    // Internal
+    //
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl SRC5: SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of InternalTrait<TContractState> {
+        /// Initializes the contract by setting default royalty.
+        fn initializer(
+            ref self: ComponentState<TContractState>,
+            default_receiver: ContractAddress,
+            default_royalty_fraction: u256
+        ) {
+            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
+            src5_component.register_interface(IERC2981_ID);
+
+            self._set_default_royalty(default_receiver, default_royalty_fraction)
+        }
+
+
+        /// The denominator with which to interpret the fee set in {_set_token_royalty} and
+        /// {_set_default_royalty} as a fraction of the sale price.
+        /// Defaults to 10000 so fees are expressed in basis points
+        fn _fee_denominator(self: @ComponentState<TContractState>) -> u256 {
+            10000
```

Comment:
```suggestion
            10_000
```

Also makes sense to move this value to a constant

Reply:
Code updated